gen-ai-rag-vertex-ai-vector-search/ingestion.tf

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Aspects to setup ingestion from Cloud Storage bucket to processing job.
+
+resource "google_storage_bucket" "ingest" {
+  name     = "ingest-${local.unique_str}"
+  location = var.region
+}
+
+
+resource "google_pubsub_topic" "ingest" {
+  name = "ingest-${local.unique_str}"
+
+  ingestion_data_source_settings {
+    cloud_storage {
+      bucket = google_storage_bucket.ingest.name
+    }
+  }
+}
+
+# Allow the Pub/Sub service account the ability to publish messages
+resource "google_project_iam_member" "pubsub" {
+  project = var.project_id
+  role    = "roles/pubsub.publisher"
+  member  = "serviceAccount:service-${data.google_project.default.project_id}@gcp-sa-pubsub.iam.gserviceaccount.com"
+}
+
+
+# Allow the Pub/Sub service account permissions to access the bucket
+resource "google_storage_bucket_iam_member" "pubsub" {
+  bucket = google_storage_bucket.ingest.name
+  role   = "roles/storage.admin"
+  member = "serviceAccount:service-${data.google_project.default.project_id}@gcp-sa-pubsub.iam.gserviceaccount.com"
+}
+
+
+resource "google_pubsub_subscription" "ingest-processing" {
```

Comment:
thoughts on adding a dead letter policy 

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_subscription#example-usage---pubsub-subscription-dead-letter

might not be helpful as it might not be in the architecture diagram, just thinking from a recommended practice whether it makes sense.

Reply:
This is a good point; the reference architecture mentions failure handling through retries, so I'll implement that.